### PR TITLE
Move logic in CoinType+Address.swift to C++ side

### PR DESCRIPTION
**Pull request by johnnynanjiang**
## Description
* implement code generation for `hrp` in `CoinType.swift`
* refactor address() in CoinType.swift to use the existing validation `TW::validateAddress()` in `Coin.cpp`

### CODE REVIEW

**Comment by johnnynanjiang** [comment]
It is working in progress, have a couple of question regarding IOST and Zilliqa address validation

1. There are 2 forms of address for Zilliqa, the current validation caters for the latter `zil`
```
"0x7FCcaCf066a5F26Ee3AFfc2ED1FA9810Deaa632C"
"zil10lx2eurx5hexaca0lshdr75czr025cevqu83uz"
```
might need another validation for `0x`, what is your thought @hewigovens @vikmeup ?

2. IOST address validation caters for a special form of address called account, like `12345xusong`, doesn't cater for derived address like `EKRQPgX7nKt8hJABwm9m3BKWGj7kcSECkJnCBauHQWin`, so I made derived address valid always.

File: src/IOST/Account.cpp [hunk 1]
```diff
@@ -4,18 +4,26 @@
...
+    for (char ch : account) {
         if ((ch < 'a' || ch > 'z') && (ch < '0' || ch > '9') && ch != '_') {
             return false;
         }
     }
     return true;
 }
 
+bool Account::isValid(const std::string& s) {
+    if (s.size() == 44) {
```
Comment (hewigovens) [inline src/IOST/Account.cpp]: just remove this check

File: swift/Sources/Addresses/CoinType+Address.swift [hunk 2]
```diff
@@ -60,8 +63,8 @@ public extension CoinType {
         case .iocoin:
             return IocoinAddress(string: string)
         case .groestlcoin:
```
Comment (hewigovens) [inline swift/Sources/Addresses/CoinType+Address.swift]: we can merge this into `.bitcoin` case

File: swift/Sources/Addresses/CoinType+Address.swift [hunk 3]
```diff
@@ -88,75 +91,10 @@ public extension CoinType {
         case .ark:
             return ARKAddress(string: string)
         case .ravencoin:
```
Comment (hewigovens) [inline swift/Sources/Addresses/CoinType+Address.swift]: merge this into `.dash` case

File: src/IOST/Account.cpp (hunk 1 above)
Comment (johnnynanjiang) [reply src/IOST/Account.cpp]: remove this check will cause CoinAddressDerivationTests() to fail for IOST, as address validation will fail to derived address (e.g. "EKRQPgX7nKt8hJABwm9m3BKWGj7kcSECkJnCBauHQWin")

File: swift/Sources/Addresses/CoinType+Address.swift (hunk 2 above)
Comment (johnnynanjiang) [reply swift/Sources/Addresses/CoinType+Address.swift]: `.groestlcoin` and `.bitcoin` are a bit different

```
            if let segwitAddress = SegwitAddress(string: string) {
                return segwitAddress
            } else {
                return GroestlcoinAddress(string: string)
            }
```

Comment (hewigovens) [reply swift/Sources/Addresses/CoinType+Address.swift]: oh ok

File: src/IOST/Account.cpp (hunk 1 above)
Comment (hewigovens) [reply src/IOST/Account.cpp]: let's check if it's a valid base58 string then

**Pull request merged**